src/org/usfirst/frc/team2585/robot/Commands.java

```diff
@@ -59,31 +59,62 @@ private static void stop() {
 	
 	
 	/**
-	 * Autonomous command that that drives the robot forward and then turns left and continues driving forward
+	 * Autonomous command that drives according to its position and the side of its switch
 	 */
 	public class Main implements AutonomousCommand {
+		private static final int timeToDriveStraight = 3100;
+		
+		private void runFromMiddleToLeft(long timeElapsed) {
+		}
+		
+		private void runFromMiddleToRight(long timeElapsed) {
+		}
 		
-		private void runFromMiddle() {
+		private void runFromLeftSide(long timeElapsed) {
 		}
 		
-		private void runFromSide() {
+		private void runFromRightSide(long timeElapsed) {
+		}
+		
+		private void runStraight(long timeElapsed) {
+			if (timeElapsed < timeToDriveStraight) {
+				driveForward();
+			} else {
+				stop();
+			}
 		}
 		
 		/* (non-Javadoc)
 		 * @see org.usfirst.frc.team2585.AutonomousCommand#execute(long)
 		 */
 		@Override
 		public void execute(long timeElapsed) {
-			if (location == 2) {
-				runFromMiddle();
+			if (location == 2 && gameData.charAt(0) == 'R') {
```

Comment:
here we should check what the location is and then call either runFromMiddle(), runFromSide(), or just runStraight if some data is missing. We can leave the gameData checking to be implemented in the methods themselves.